DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/ip_authoring_mmhost/README.md

```diff
@@ -0,0 +1,368 @@
+If no memory-mapped interface macro is specified for a pointer kernel argument, then a mmhost interface with all default parameters is generated for it.
+```c++
+struct PointerIP{
+  int *x, *y, *z;
+  int size;
+
+    void operator()() const {
+      for (int i = 0; i < size; ++i) {
+        int mul = x[i] * y[i];
+        int add = mul + z[i];
+        y[i] = mul;
+        z[i] = add;
+      }
+    }
+};
+```
+
+### Memory-mapped interface implementation restriction
+
+The mmhost macros are restricted to use with functors, which means that they cannot be used when the kernel is written as a lambda function. Also, all arguments of the macro must be specified, even if just one parameter is being changed from the default values.
+
+#### Example 1: A kernel expressed using a functor model with mmhost macro.
+```c++
+struct MyIP {
+  mmhost(
+    1,       // buffer_location or aspace
+    28,      // address width
+    64,      // data width
+    16,      // latency
+    1,       // read_write_mode, 0: ReadWrite, 1: Read, 2: Write
+    1,       // maxburst
+    0,       // align, 0 defaults to alignment of the type
+    1        // waitrequest, 0: false, 1: true
+  ) int *my_pointer;
+
+  void operator()() const {
+    ...
+    // Kernel code
+    my_pointer[x] = y;
+    ...
+  }
+}
+```
+
+
+
+An Avalaon bus is generate for every unique global memory in the system. 
+
+If the mmhost argument is specified as a conduit interface then an input wire is generated at the top-level device image IP to carry the the pointer address into the kernal. 
+
+If the mmhost argument is specified as a register-mapped argument, the input pointer is written into a register map associated with the kernal. An Avalon bus is generated for writing those arguments into a register map, with no seperate wire needed at the top level device image to input the address.
+#### Example 2: Changing default mmhost interface. 
+```c++
+// A memory-mapped interface that is implemented as a register.
+  register_map_mmhost(
+    1,       // buffer_location or aspace
+    28,      // address width
+    64,      // data width
+    16,      // latency
+    1,       // read_write_mode, 0: ReadWrite, 1: Read, 2: Write
+    1,       // maxburst
+    0,       // align, 0 defaults to alignment of the type
+    1        // waitrequest, 0: false, 1: true
+  ) int *memory_mapped_pointer;
+// A memory-mapped interface that is implemented as a wire.
+  conduit_mmhost(
+    1,       // buffer_location or aspace
+    28,      // address width
+    64,      // data width
+    16,      // latency
+    1,       // read_write_mode, 0: ReadWrite, 1: Read, 2: Write
+    1,       // maxburst
+    0,       // align, 0 defaults to alignment of the type
+    1        // waitrequest, 0: false, 1: true
+  ) int *memory_mapped_pointer;
+```
+
+If unspecified, when using only the ```mmhost(...)``` macro, a register_map based input argument interface is used.
+
+### Memory Base Addresses assigned by the compiler
+
```

Comment:
We have decided against drawing attention to the ipinterfaces.xml file and instead want the users to understand the feature from the HTML reports alone.

So here, state that the inferred memory and its properties can be seen in the Memory Viewer (get the exact panel names from the HTML report panels)